src/herder/simulation/SimulationTxSetFrame.cpp

```diff
@@ -0,0 +1,121 @@
+// Copyright 2019 Stellar Development Foundation and contributors. Licensed
+// under the Apache License, Version 2.0. See the COPYING file at the root
+// of this distribution or at http://www.apache.org/licenses/LICENSE-2.0
+
+#include "herder/simulation/SimulationTxSetFrame.h"
+#include "crypto/SHA.h"
+#include "transactions/simulation/SimulationTransactionFrame.h"
+#include "xdrpp/marshal.h"
+#include <numeric>
+
+namespace stellar
+{
+
+static Hash
+computeContentsHash(Hash const& previousLedgerHash,
+                    std::vector<TransactionEnvelope> transactions)
+{
+    std::sort(
```

Comment:
it's a bit unfortunate that we're duplicating a lot of code from `TxSetFrame`, can we instead just have a `TxSetFrame` member that we use to delegate all those things?

Reply:
yeah 3 seems fine for now, the performance impact of it should go away when #2163 is done

Reply:
Done.